src/main/java/org/opensearch/flowframework/workflow/WorkflowProcessSorter.java

```diff
@@ -244,11 +254,17 @@ public void validateGraph(List<ProcessNode> processNodes, WorkflowValidator vali
                 );
             }
         }
+    }
 
+    private WorkflowValidator readWorkflowValidator() throws IOException {
+        return WorkflowValidator.parse("mappings/workflow-steps.json");
     }
 
-    private TimeValue parseTimeout(WorkflowNode node) {
-        String timeoutValue = (String) node.userInputs().getOrDefault(NODE_TIMEOUT_FIELD, NODE_TIMEOUT_DEFAULT_VALUE);
+    private TimeValue parseTimeout(WorkflowNode node) throws IOException {
+        WorkflowValidator validator = readWorkflowValidator();
+        String nodeTimeoutValue = Optional.ofNullable(validator.getWorkflowStepValidators().get(node.type()).getTimeout())
+            .orElse(NODE_TIMEOUT_DEFAULT_VALUE);
```

Comment:
I really like this bit of code for handling the possible null I mentioned needing handling in an earlier comment.

That said, this implies we can/should leave the timeout value blank in the json if we want the default, but you've added this same default (10s) to every node, so it's a bit inconsistent.

Reply:
As mentioned in this issue - https://github.com/opensearch-project/flow-framework/issues/249, I added 15s for Deploy model step, and 10s for the rest steps. For the new revision, according to your [comment above](https://github.com/opensearch-project/flow-framework/pull/317#discussion_r1435777193) I changed the timeout from 10s to 60s for `register_local_model` step :)

Reply:
While writing docs for the settings, I see a timeout setting that we have that seems to only apply to the Create Workflow API.  This seems silly as we already have an OpenSearch timeout for that.   But we probably should have a setting for the default workflow step timeout (or use that exact setting)... unless we are using it and I'm confused!  :|